salt/minion.py

```diff
@@ -585,9 +585,10 @@ def __init__(self, opts):
         # Clean out the proc directory (default /var/cache/salt/minion/proc)
         if (self.opts.get('file_client', 'remote') == 'remote'
                 or self.opts.get('use_master_when_local', False)):
-            if HAS_ZMQ:
-                zmq.eventloop.ioloop.install()
-            io_loop = LOOP_CLASS.current()
+            if self.opts['transport'] == 'zeromq' and HAS_ZMQ:
+                io_loop = zmq.eventloop.ioloop.ZMQIOLoop()
+            else:
+                io_loop = LOOP_CLASS.current()
```

Comment:
right, and I think this could be cleaner. The main thing is that the zmq install line seemed to be causing an issue when the loop already exists. So I think we are in a good place, but I do think that we should gate these by transport rather than availability, we don't want to create zmq loop if we have transport set to tcp and just happen to have pyzmq installed